paper/src/main/java/vg/civcraft/mc/citadel/listener/ActivityListener.java

```diff
@@ -0,0 +1,49 @@
+package vg.civcraft.mc.citadel.listener;
+
+import org.bukkit.Location;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
+import org.bukkit.event.Listener;
+import org.bukkit.event.player.PlayerJoinEvent;
+import org.bukkit.event.player.PlayerMoveEvent;
+import org.bukkit.event.world.ChunkLoadEvent;
+import org.bukkit.event.world.ChunkUnloadEvent;
+import vg.civcraft.mc.citadel.Citadel;
+import vg.civcraft.mc.citadel.activity.ActivityMap;
+
+public class ActivityListener implements Listener {
+
+	private final ActivityMap map;
+
+	public ActivityListener(ActivityMap map) {
+		this.map = map;
+	}
+
+	@EventHandler
+	public void onPlayerMove(PlayerMoveEvent event) {
+		Location from = event.getFrom();
+		Location to = event.getTo();
+		//noinspection ConstantConditions
+		if (to == null || from == null) {
+			return;
+		}
+
```

Comment:
To limit the "spam" of this event from stuff such as a slight press, I'd recommend a check roughly here on the event method `hasChangedBlock()`

Reply:
Ok, thanks, will look it later today

Reply:
Done: https://github.com/CivMC/Citadel/pull/39/commits/b83610d297d9730adf41d1a3bafb6d8ddfdd4de1#diff-3ed12afc97b01c114b125022d2eec99cf5b3b205e7105cbfe911f7ba044bd6a9R26